shephard-core/src/main/java/com/devsda/platform/shephardcore/service/ExecutionDocumentService.java

```diff
@@ -0,0 +1,100 @@
+package com.devsda.platform.shephardcore.service;
+
+import com.devsda.platform.shephardcore.model.ShephardConfiguration;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.inject.Inject;
+import com.mongodb.MongoClient;
+import com.mongodb.MongoClientURI;
+import com.mongodb.MongoWriteException;
+import com.mongodb.client.MongoCollection;
+import com.mongodb.client.result.UpdateResult;
+import org.bson.Document;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Date;
+import java.util.Map;
+
+public class ExecutionDocumentService {
+
+    @Inject
+    private ShephardConfiguration shepherdConfiguration;
+
+    private MongoClient mongoClient;
+
+    private static final Logger log = LoggerFactory.getLogger(ExecutionDocumentService.class);
+
+    public boolean insertExecutionDetails(Integer executionId, Map<String, Object> initialPayload) {
+        this.mongoClient = getMongoClient();
+        try {
+            ObjectMapper mapper = new ObjectMapper();
+            mapper.configure(com.fasterxml.jackson.databind.SerializationFeature.WRITE_DATES_AS_TIMESTAMPS , false);
+            String jsonString = "";
+            try {
+                jsonString = mapper.writeValueAsString(initialPayload);
+            }catch(JsonProcessingException e){
+                log.error(String.format("Unable to Process the initial payload for execution id : %s.", executionId), e);
+                return false;
+            }
+            MongoCollection<Document> collection = ExecutionDocumentServiceHelper.getMongoCollection(this.mongoClient, this.shepherdConfiguration.getDataSourceDetails().getDbname(), this.shepherdConfiguration.getDataSourceDetails().getCollectionname());
+
+            if (collection != null) {
```

Comment:
I think, we should block one field in doocument for Shepherd Metadata.

Name can be : "shepherd_metadata" : { "execution_id" : 12, "client_id" : 14, etc, etc}.

Reply:
added TODO